File: Python/ArraysAndHashing/TwoSum.py
def twoSum(nums: list[int], target: int) -> list[int]:
    value_holder = {}

    for i in range(len(nums)):
        needed = target - nums[i]

        if needed in value_holder:
            return [value_holder[needed], i]

        value_holder[nums[i]] = i

    return []

File: TypeScript/ArraysAndHashing/TwoSum.ts
function twoSum(nums: number[], target: number): number[] {
    const valueHolder = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const needed = target - nums[i];

        if (valueHolder.has(needed)) {
            return [valueHolder.get(needed)!, i];
        }

        valueHolder.set(nums[i], i);
    }

    return [];
}
